base/array.jl

```diff
@@ -27,11 +27,12 @@ import Core: arraysize, arrayset, arrayref
 
 """
     Array{T}(dims)
+    Array{T,N}(dims)
```

Comment:
We use brackets in lots of other places in the documentation for optional arguments.   Can you suggest a clearer way to write the `SharedArray` doc that doesn't involve duplicating the long call-signature line?